utils/bcm27xx-eeprom/Makefile

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (C) 2020 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=bcm27xx-eeprom
+PKG_VERSION:=a9ca308223c1d0426b9ab320696b95954078c3b4
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/raspberrypi/rpi-eeprom/tar.gz/$(PKG_VERSION)?
+PKG_HASH:=072dbbd4b53c2c0b5764ff628f63cc8d679a99cfe84d8f50acea06922084515e
+
+PKG_MAINTAINER:=Álvaro Fernández Rojas <noltari@gmail.com>
+
+include $(INCLUDE_DIR)/package.mk
+
+TAR_OPTIONS:=--strip-components 1 $(TAR_OPTIONS)
+TAR_CMD=$(HOST_TAR) -C $(1) $(TAR_OPTIONS)
+
+define Package/bcm27xx-eeprom
+  SECTION:=utils
+  CATEGORY:=Utilities
+  DEPENDS:=bcm27xx-userland +blkid +pciutils +python3-light
+  TITLE:=BCM27xx EEPROM tools
+endef
+
+define Package/bcm27xx-eeprom/description
+  BCM27xx EEPROM tools.
+endef
+
+define Build/Compile
+	true
+endef
+
+define Package/bcm27xx-eeprom/install
+	$(INSTALL_DIR) $(1)/etc/default
+	$(INSTALL_CONF) $(PKG_BUILD_DIR)/rpi-eeprom-update-default $(1)/etc/default/rpi-eeprom-update
```

Comment:
`/etc/config` would be the traditional OpenWrt place for config files.

Also suggest to add a `Package/bcm27xx-eeprom/conffiles` section so it can be saved between sysupgrades.

Reply:
But the default file would need to become something UCI compatible, right?

Reply:
Not really, since nothing uci-related is reading from it.

Reply:
Okay I though etc/config is for UCI files only.

Reply:
I guess keeping the file in a directory like `/etc/rpi-eeprom` (or just in `/etc`) would also be ok. `/etc/default` is a Debian thing.

Reply:
Ok, I will move it to /etc (moving it to /etc/config is a bad idea), but it will require more patches...
BTW, I think there are some other packages on OpenWrt using /etc/default. Is this change really needed?

Reply:
@jefferyto 
FYI, having non-uci files in /etc/config can cause visible uci error messages in console when uci commands are run.

`uci: Entry not found` would be the typical error that materialises

Reply:
@hnyman Good to know, thanks.

@Noltari No, nothing is ever "really needed" - this is a question of what is better and what is worse.

If I was a Debian/Ubuntu user, I would be familiar with (or at least aware of) `/etc/default`. If I was a Fedora/Arch/etc. user, probably not.

I believe an OpenWrt user would look for a config file in this approximate order:
1. `/etc/config/<package name>`
2. `/etc/<package name>.conf`
3. `/etc/<package name>/<some files>`

If I was being really pedantic, I would suggest you add a uci config file to `/etc/config` then have `rpi-eeprom-update` read the uci values (or have an init script that converted the uci values into the current config file format). But even I realize this would be overkill.

I'm sure there are other packages that do other things, that doesn't mean what they are doing is a good idea.

Of course, this is just a suggestion - you are free to do what you want :slightly_smiling_face:

Reply:
Adding a uci config file is indeed overkill, so in my latest update I removed this file completely and patched the values directly on rpi-eeprom-update.

Reply:
Without a config file, how does the user select whether they want critical, stable, or beta firmwares?

Also, please stop marking discussions as resolved. Just because you think they are resolved doesn't mean other parties agree.

Reply:
I suggest adding the file to `/etc/bcm27xx-eeprom.conf` and patch the script to read this file. Are you both okay with this?

Reply:
@jefferyto Ok, I won't mark any discussion as resolved until you give your OK. Can you start being more nice and less unfriendly?
The user is still able to select the configuration options by exporting them to the env or:
```FIRMWARE_RELEASE_STATUS="beta" rpi-eeprom-update```

@aparcar I just want an agreement that lets me submit a final version of this package that can be merged, so I will adapt to any directives from you or @jefferyto.

Reply:
I think a having a config is better but that's just my point of view, @jefferyto has much more experience as a maintainer than me!

Reply:
@aparcar done

Reply:
While I don't think I was being intentionally unfriendly, I wasn't being intentionally friendly either. My comments were and are directed at the work, not to you personally.

I'll note all of my comments have been suggestions, most even use the word "suggest". I am not an official member of this project; my suggestions do not need to be adopted, my approval of the changes is not required.

If you will pardon a slight digression: I think a common problem with developers and open source in general is a lack of attention to user experience, or even just user friendliness. I don't blame you personally; we like working with computers and not necessarily with users, that's why we do what we do. I believe though if we want to make things for other people to use, then we should also think about how they will use the things we make.

I'm sure this package works fine on the technical level. Almost all of my suggestions have had to do with making it more user-friendly.

I see you have brought back the config file. I also see the changes to the script's default values have been removed, the Makefile does not have a `conffiles` section, the config file name has not been changed in the script's [usage](https://github.com/raspberrypi/rpi-eeprom/blob/a9ca308223c1d0426b9ab320696b95954078c3b4/rpi-eeprom-update#L326) [message](https://github.com/raspberrypi/rpi-eeprom/blob/a9ca308223c1d0426b9ab320696b95954078c3b4/rpi-eeprom-update#L348).

I don't want to speculate on why this is the way it is. I do think my usefulness on this PR has probably come to an end, so I will bow out of the discussion :smile: :pray: